presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -0,0 +1,208 @@
+                log.error(errorMessage);
+                response.setStatus(SC_INTERNAL_SERVER_ERROR);
+                response.setHeader(CONTENT_TYPE, PRESTO_PAGES);
+                response.setHeader(PRESTO_TASK_INSTANCE_ID, taskManager.getTaskInstanceId(taskId));
+                response.setHeader(PRESTO_PAGE_TOKEN, String.valueOf(token));
+                response.setHeader(PRESTO_PAGE_NEXT_TOKEN, String.valueOf(token));
+                response.setHeader(PRESTO_BUFFER_COMPLETE, String.valueOf(false));
+                response.sendError(SC_INTERNAL_SERVER_ERROR, errorMessage);
+            }
+        });
+
+        ListenableFuture<BufferResult> bufferResultFuture = taskManager.getTaskResults(taskId, bufferId, token, maxSize);
+        bufferResultFuture = addTimeout(
+                bufferResultFuture,
+                () -> BufferResult.emptyResults(taskManager.getTaskInstanceId(taskId), token, false),
+                waitTime,
+                timeoutExecutor);
+
+        ServletOutputStream out = response.getOutputStream();
+        addCallback(bufferResultFuture, new FutureCallback<BufferResult>()
+                {
+                    @Override
+                    public void onSuccess(BufferResult bufferResult)
+                    {
+                        Queue<SerializedPage> serializedPages = new LinkedList<>(bufferResult.getSerializedPages());
+
+                        response.setHeader(CONTENT_TYPE, PRESTO_PAGES);
+                        response.setHeader(PRESTO_TASK_INSTANCE_ID, bufferResult.getTaskInstanceId());
+                        response.setHeader(PRESTO_PAGE_TOKEN, String.valueOf(bufferResult.getToken()));
+                        response.setHeader(PRESTO_PAGE_NEXT_TOKEN, String.valueOf(bufferResult.getNextToken()));
+                        response.setHeader(PRESTO_BUFFER_COMPLETE, String.valueOf(bufferResult.isBufferComplete()));
+
+                        if (serializedPages.isEmpty()) {
+                            response.setStatus(SC_NO_CONTENT);
+                            asyncContext.complete();
+                        }
+                        else {
+                            int pageMetadataSize = SIZE_OF_INT * 3 + SIZE_OF_BYTE;
+                            int contentLength = serializedPages.stream()
+                                    .mapToInt(page -> page.getSizeInBytes() + pageMetadataSize)
+                                    .sum();
+                            response.setHeader(CONTENT_LENGTH, String.valueOf(contentLength));
+                            out.setWriteListener(new SerializedPageWriteListener(serializedPages, asyncContext, out));
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Throwable thrown)
+                    {
+                        String errorMessage = format("Error getting task result from TaskManager for request %s : %s, ", requestURI, thrown.getMessage());
```

Comment:
Please include this message by doing `response.sendError(SC_INTERNAL_SERVER_ERROR, errorMessage);`